### ISSUE TITLE
Switch skaffold to build with go 1.1.9

### ISSUE THREAD

**Issue by tejal29** [GoogleContainerTools/skaffold#7944]
This is a P0 and should happen before GA release.

Looking into CVE-2019-19814  and CVE-2021-3973. 
Is skaffold affected by them?
Yes -> switch skaffold to v1.19 before GA.

No -> Why not?

**Comment by ericzzzzzzz** [comment]
TestCacheAPITriggers is failing. 
I think it's safe to say that the functionality we're testing with this test is OK.. 
but the weird thing is that most of the time the test gets stuck even the test finishes if it run on go 1.19.2 test runner.. it seems related to a skaffold helper function for running a backgound skaffold process..
commenting this out can make test pass... https://github.com/GoogleContainerTools/skaffold/blob/main/integration/skaffold/helper.go#L243-L246 
still don't know why this is happening..

**Comment by briandealwis** [comment]
@ericzzzzzzz  are you sure it’s blocking on that `cmd.Wait()` and not the [other `cmd.Wait()` further down](https://github.com/GoogleContainerTools/skaffold/blob/b5ce32043047b91205b398d9a1b4b539d4c673c4/integration/skaffold/helper.go#L252)? I don’t see why a test would hang because of that goroutine.

If the hang is the later ˋcmd.Wait()ˋ then the cause is likely a test calling [`RunLive`](https://github.com/GoogleContainerTools/skaffold/blob/1f5af29273e2938c66311fcc22ddf0fcceeeffd7/integration/skaffold/helper.go#L208) **but is not reading from the returned `io.Reader`**, which is an `io.Pipe`. On Unix-like OS, the OS will only buffer a certain amount of data written to a pipe and then block the producing process if it’s not consumed. See #5083 for details(and #5220 for an example where we fixed a hang by using by buffets instead of a pipe).

I can see here that `TestCustomTest` uses `RunLive` but is not consuming the output pipe:
https://github.com/GoogleContainerTools/skaffold/blob/338d40d22afaf6a695ccc543cc1e4162d57078aa/integration/custom_test.go#L47

**Comment by ericzzzzzzz** [comment]
@briandealwis  ohh.. I was just saying that part might be related,  something changed in cmd.Wait() method between go18 and go19, the change may lead to some race condition when multiple goroutines call cmd.Wait() method, in this test.. it seems that the one in sub goroutine is able to block the cmd.Wait() method in test goroutine.  The following code also hangs on my machine sometimes. 

```go
	MarkIntegrationTest(t, CanRunWithoutGcp)

	ns, _ := SetupNamespace(t)

	// Disable caching to ensure we get a "build in progress" event each time.
	skaffold.Dev("--cache-artifacts=false").InDir("examples/getting-started").InNs(ns.Name).RunBackground(t)
	time.Sleep(5 * time.Second)

```

**Comment by gsquared94** [comment]
fixed via #7871
